src/presentation/test/integration/registration/generalPartner/address/postcode-general-partner-principal-office-address.test.ts

```diff
@@ -0,0 +1,95 @@
+import request from "supertest";
+
+import enTranslationText from "../../../../../../../locales/en/translations.json";
+import cyTranslationText from "../../../../../../../locales/cy/translations.json";
+
+import { appDevDependencies } from "../../../../../../config/dev-dependencies";
+import app from "../../../app";
+
+import { getUrl, setLocalesEnabled, toEscapedHtml, testTranslations } from "../../../../utils";
+import { POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL } from "presentation/controller/addressLookUp/url";
+import GeneralPartnerBuilder, {
+  generalPartnerLegalEntity
+} from "../../../../builder/GeneralPartnerBuilder";
+import AddressPageType from "../../../../../controller/addressLookUp/PageType";
+
+describe("Postcode general partner's principal office address page", () => {
+  const URL = getUrl(POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL);
+
+  beforeEach(() => {
+    setLocalesEnabled(false);
+    appDevDependencies.cacheRepository.feedCache(null);
+
+  });
+
+  describe("Get postcode general partner's principal office address page", () => {
+    it("should load the principal office address page with English text", async () => {
+      setLocalesEnabled(true);
+
+      const generalPartner = new GeneralPartnerBuilder()
+        .withId(appDevDependencies.generalPartnerGateway.generalPartnerId)
+        .isPerson()
```

Comment:
for this screen with principal office address we should use legal entity general partner instead of person. A GP Person will have a usual residential address and a legal entity will have a principal office address

Reply:
Changed to specify legal entity.